test/gpt2/tmodel.m

```diff
@@ -7,25 +7,24 @@
         model = @gpt2.model
     end
     
+    properties(TestParameter)
+        InputData = iGetInputData();
+    end
+    
     methods(Test)
-        function canUseModel(test)
-            inputs = test.prepareInputs();
-            test.verifyWarningFree(@() test.model(inputs{:}));
+        function canUseModel(test, InputData)
```

Comment:
I'd like to see a new test that checks each batched operation matches calling the operation on each observation in turn, i.e. `f([x1,x2]) = [f(x1),f(x2)]` in some sense.

Doing that at the model level will be most efficient, but it'd be good practice to do that at the unit level too. I can take a look at this too, but I think I'd rather see it before merging into master.

Reply:
Absolutely. I was using such a test locally to confirm my changes. I'll add something to `tmodel`.